src/plotApplications/helpers.js

```diff
@@ -309,45 +331,88 @@ export const prepareApplicationForSubmission = () => {
     'formEntries.fileFieldIds'
   );
 
-  const attachMeta = (rootLevelSections) => {
-    Object.keys(rootLevelSections).forEach((sectionName) => {
-      switch (sectionName) {
-        case APPLICANT_SECTION_IDENTIFIER:
-          rootLevelSections[sectionName].forEach((applicant) => {
-            if (applicant.fields['hakija'] === 'Yritys') {
-              applicant.metadata = {
-                identifier: null // todo
-              };
-            } else {
-              applicant.metadata = {
-                identifier: null // todo
+  try {
+    const attachMeta = (rootLevelSections) => {
+      Object.keys(rootLevelSections).forEach((sectionName) => {
+        const section = rootLevelSections[sectionName];
+
+        switch (sectionName) {
+          case APPLICANT_SECTION_IDENTIFIER:
+            section.forEach((applicant) => {
+              applicant.sections = _.pickBy(
+                _.cloneDeep(applicant.sections),
+                (subsection, sectionIdentifier: string) => [ApplicantTypes.BOTH, applicant.metadata.applicantType].includes(
+                  applicant.sectionRestrictions[sectionIdentifier]
+                )
+              );
+
+              try {
+                let identifier;
+                if (applicant.metadata.applicantType === ApplicantTypes.PERSON) {
```

Comment:
Not if written like that (since there's no overlap between the allowed values for applicantType and the section identifiers) but I think I get what you mean. Rather than defining simple string constants, I could do

```js
export const APPLICANT_MAIN_IDENTIFIER_SECTIONS = {
  [ApplicantTypes.COMPANY]: 'yrityksen-tiedot',
  [ApplicantTypes.PERSON]: 'henkilon-tiedot'
};

export const APPLICANT_MAIN_IDENTIFIER_FIELDS = {
  [ApplicantTypes.COMPANY]: 'y-tunnus',
  [ApplicantTypes.PERSON]: 'henkilotunnus'
}
```

and then do away with this if-else by just writing

```js
                  const sectionWithIdentifier = applicant.sections[APPLICANT_MAIN_IDENTIFIER_SECTIONS[applicant.metadata.applicantType]];
                  identifier = sectionWithIdentifier?.fields[APPLICANT_MAIN_IDENTIFIER_FIELDS[applicant.metadata.applicantType]]
                    || sectionWithIdentifier[0]?.fields[APPLICANT_MAIN_IDENTIFIER_FIELDS[applicant.metadata.applicantType]];
```
here instead. (With some additional intermediate variables for brevity probably. And the constant objects could also be combined even further.)